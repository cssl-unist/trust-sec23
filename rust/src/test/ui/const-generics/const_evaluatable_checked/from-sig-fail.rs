#![feature(const_generics, const_evaluatable_checked)]
#![allow(incomplete_features)]

fn test<const N: usize>() -> [u8; N - 1] {
    //~^ ERROR evaluation of constant
    todo!()
}

fn main() {
    test::<0>();
}
