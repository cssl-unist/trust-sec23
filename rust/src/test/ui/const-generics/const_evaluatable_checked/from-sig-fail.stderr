error[E0080]: evaluation of constant value failed
  --> $DIR/from-sig-fail.rs:4:35
   |
LL | fn test<const N: usize>() -> [u8; N - 1] {
   |                                   ^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow

error: aborting due to previous error

For more information about this error, try `rustc --explain E0080`.
